sklearn/mixture/gmm.py

```diff
@@ -501,10 +523,33 @@ def fit(self, X, y=None):
             self.covars_ = best_params['covars']
             self.means_ = best_params['means']
             self.weights_ = best_params['weights']
+        else:
```

Comment:
I added it because there was a comment in the doc string about setting n_iter=0 if you just wanted to initialize the model. I read somewhere else that this was needed for an HMM to work. 
Good suggestion about adding a test, thanks.